src/App.tsx

```diff
@@ -60,8 +67,14 @@ const App: React.FC = () => {
         .then(result => result.json())
         .then(resultJSON => {
           if (resultJSON.shortValue) {
-            const cardData = JSON.parse(resultJSON.shortValue) as CardData
-            processCardData(cardData, resultJSON.longValueExists)
+            if (resultJSON.present) {
+              const cardData = JSON.parse(resultJSON.shortValue) as CardData
+              processCardData(cardData, resultJSON.longValueExists)
+            } else {
+              setIsWritableCard(false)
+            }
+          } else {
+            setIsWritableCard(true)
```

Comment:
I'm struggling to follow the logic in these nested if blocks. I think it's the `resultJSON.present`… does `present` mean "card is present in reader"? Renaming `resultJSON` to `card` will help.

Ah, the part that doesn't make sense: if `resultJSON.shortValue` exists, then won't the nested `resultJSON.present` always be true?

Reply:
you're correct, the logic is slightly wrong here. I think the API still makes sense, you're reading the card, and it tells you whether it's present or not. But of course that inner `if` should be outside. I'll fix. Good catch